tasks/gulp/__tests__/after-build-package.test.js

```diff
@@ -25,83 +31,77 @@ describe('package/', () => {
     componentNames = [...(await getDirectories(configPaths.components)).keys()]
   })
 
-  it('should contain the expected files', () => {
-    // Build an array of the files that are present in the package directory.
-    const actualPackageFiles = () => {
-      return recursive(configPaths.package).then(
-        files => {
-          return files
-            // Remove /package prefix from filenames
-            .map(file => file.replace(/^package\//, ''))
-            // Sort to make comparison easier
-            .sort()
-        },
-        error => {
-          console.error('Unable to get package files', error)
-        }
-      )
-    }
-
-    // Build an array of files we expect to be found in the package directory,
-    // based on the contents of the src directory.
-    const expectedPackageFiles = () => {
-      const filesToIgnore = [
-        '.DS_Store',
-        '*.test.*',
-        '*.yaml',
-        '*.snap',
-        '*/govuk/README.md'
-      ]
-
-      const additionalFilesNotInSrc = [
-        'package.json',
-        'govuk-prototype-kit.config.json',
-        '**/macro-options.json',
-        '**/fixtures.json',
-        'README.md'
-      ]
-
-      return recursive(configPaths.src, filesToIgnore).then(
-        files => {
-          let filesNotInSrc = files
-          // Use glob to generate an array of files that accounts for wildcards in filenames
-          filesNotInSrc = glob.sync('{' + additionalFilesNotInSrc.join(',') + '}', { cwd: 'package' })
-
-          return files
-            .map(file => {
-              // Remove /src prefix from filenames
-              const fileWithoutSrc = file.replace(/^src\//, '')
-
-              // Account for govuk-esm folder
-              if (fileWithoutSrc.split('.').pop() === 'mjs') {
-                const esmFile = fileWithoutSrc.replace('govuk/', 'govuk-esm/')
-                const umdFile = fileWithoutSrc.replace('.mjs', '.js')
-
-                return [umdFile, esmFile]
-              } else {
-                return fileWithoutSrc
-              }
-            })
-            // Allow for additional files that are not in src
-            .concat(filesNotInSrc)
-            .flat()
-            // Sort to make comparison easier
-            .sort()
-        },
-        error => {
-          console.error('Unable to get package files', error)
+  it('should contain the expected files', async () => {
+    const filterPatterns = [
+      '!**/.DS_Store',
+      '!**/*.test.*',
+      '!**/__snapshots__/',
+      '!**/__snapshots__/**',
+      `!${configPaths.src}README.md`
+    ]
+
+    // Build array of expected output files
+    const filesExpected = [...listingSource]
+      .flatMap(listingToArray)
+
+      // Apply filters
```

Comment:
In JavaScript "filter" means "keep it in the array" so I'm going to avoid **Filter out** as really it should be **Filter in** and then "shouldn't" flips it again 😆 

I'll keep it generic:

```suggestion
      // Only include files matching filter patterns
```

Reply:
If we want to avoid "Filter", could we go for "Exclude files in the source listing that shouldn't be in the package"?

A generic comment like that last suggestion doesn't bring much to the table when you read the file, as this is just a translation of `.filter(filterPath(filterPatterns))`, and with the addition of the helper, it'll already kind of clear. 

What's less clear and I wanted to highlight in the comment is the role of that computation in checking the package, which is to weed out from the expected files those that we know won't be in the package for sure.

Sorry if it sounds nitpicky, but I think having such a trail will help the next person reading that file 😄

Reply:
@romaricpascal I've removed the comment (saves this being a blocker)

If there's something in the PR that's less understandable than it was before let me know 🙈